Update Transformation Template

HaSistrunk opened the pull request:
Update Cloud Formation template for the [transformation service](https://github.com/RockefellerArchiveCenter/digital_ingest_transformation) (formerly Aquarius). 

- Renamed from Aquarius to Transformation
- Added Dev as default for application environment parameter
- Added SSM Role resource
- Updated parameters and resources: removed unnecessary holdovers from copying the Discovery template
- Added `EventsTopicArn` to nested parameters

Question: is Package_ID needed here? We weren't 100% on how that is being handled.

HaSistrunk committed:
fix typo

HaSistrunk committed:
update aquarius name and add some annotations

p-galligan committed:
set username

Explicitly sets username for transformationuser

p-galligan committed:
Update template

Adds sns policy, environment variables, and parameters where needed.

Still unsure about package_id

p-galligan committed:
Updating env

Updates env variables

p-galligan committed:
Update parameter names

Update parameter names to keep with conventions

HaSistrunk committed:
add todo comment for TransformationTask nested stack

HaSistrunk committed:
use correct filename

p-galligan committed:
Clean up template

Removes unnecessary params and splits ssm and sns roles.

p-galligan committed:
Clean up params

Removes unused params/variables. Fixes a couple of variable names

HaSistrunk committed:
cleanup, add parameter to nested stack, question package id

p-galligan committed:
Remove unnecessary param

Removes sns topic param and correct policy name capitalizations.

p-galligan committed:
Merge branch 'transformation_cloudformation' of https://github.com/RockefellerArchiveCenter/digital_ingest_pipeline into transformation_cloudformation

p-galligan committed:
Update params

Update param names

helrond commented on digital_ingest_transformation_template.yaml:
```diff
@@ -75,108 +57,126 @@ Resources:
-  AquariusUser:
+  TransformationUser:
     Type: AWS::IAM::User
+    UserName: TransformationUser
```
Let's talk about this - I have typically not assigned usernames because names can cause problems with redeployments. See also the warning in the `UserName` property here: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-user.html

p-galligan replied:
Per discussion, we can get rid of this because it's not adding anything to our use cases. The PhysicalID string isn't necessary.

helrond commented on digital_ingest_transformation_template.yaml:
```diff
@@ -28,35 +29,16 @@ Parameters:
-  DigitizationPath:
-    Type: String
-    Description: Path to deliver packages of digitized content for further processing.
-  DigitizationUrl:
-    Type: String
-    Description: URL to post to to trigger further processing for packages of digitized content.
+    Description: String representation of the package identifier
```
This is not needed! The lambda which starts the instance of the task supplies this value, since it will be unique to that instance.

helrond commented on digital_ingest_transformation_template.yaml:
```diff
@@ -75,108 +57,126 @@ Resources:
+              Value: !GetAtt TransformationSsmRole.Arn
             - Name: AWS_SNS_TOPIC
               Value: !Ref EventsTopicArn
+            #Is this needed?
+            - Name: PACKAGE_ID
+              Value: !Ref PackageID
```
Also not needed!

p-galligan committed:
Remove unnecessary params

Removes references to package_id and removes the username from TransformationUser.
